contentcuration/contentcuration/frontend/channelEdit/components/edit/CompletionOptions.vue

```diff
@@ -0,0 +1,797 @@
+          this.currentCompletionDropdown === CompletionDropdownMap.completeDuration
+        );
+      },
+      requiresAudioVideoDefault() {
+        return this.audioVideoResource && !this.value.model;
+      },
+      completionDropdownIsCompleteDuration() {
+        return (
+          (this.completionDropdown === null &&
+            this.currentCompletionDropdown === CompletionDropdownMap.completeDuration) ||
+          (this.completionDropdown === CompletionDropdownMap.completeDuration &&
+            this.currentCompletionDropdown === null) ||
+          (this.completionDropdown === CompletionDropdownMap.completeDuration &&
+            this.currentCompletionDropdown === CompletionDropdownMap.completeDuration) ||
+          this.currentCompletionDropdown === CompletionDropdownMap.completeDuration
+        );
+      },
+      completionDropdownIsAllContentViewed() {
+        return (
+          (this.completionDropdown === null &&
+            this.currentCompletionDropdown === CompletionDropdownMap.allContent) ||
+          (this.completionDropdown === CompletionDropdownMap.allContent &&
+            this.currentCompletionDropdown === null) ||
+          (this.completionDropdown === CompletionDropdownMap.allContent &&
+            this.currentCompletionDropdown === CompletionDropdownMap.allContent)
+        );
+      },
+      durationDropdown: {
+        get() {
+          const defaultStateForAudioVideo =
+            this.value.suggested_duration === null &&
+            this.value.suggested_duration_type === null &&
+            this.audioVideoResource;
+          if (
+            this.value.model === CompletionCriteriaModels.REFERENCE ||
+            (this.currentCompletionDropdown === CompletionDropdownMap.completeDuration &&
+              this.currentDurationDropdown === DurationDropdownMap.REFERENCE)
+          ) {
+            return DurationDropdownMap.REFERENCE;
+          } else if (this.value.model === CompletionCriteriaModels.PAGES) {
+            if (this.isLongActivity) {
+              return DurationDropdownMap.LONG_ACTIVITY;
+            }
+            if (this.isShortActivity) {
+              return DurationDropdownMap.SHORT_ACTIVITY;
+            }
+            if (this.isExactTime) {
+              return DurationDropdownMap.EXACT_TIME;
+            }
+          } else if (
+            this.value.model === CompletionCriteriaModels.TIME ||
+            defaultStateForAudioVideo
+          ) {
+            return DurationDropdownMap.EXACT_TIME;
+          } else {
+            if (this.isLongActivity) {
+              return DurationDropdownMap.LONG_ACTIVITY;
+            }
+            if (this.isShortActivity) {
+              return DurationDropdownMap.SHORT_ACTIVITY;
+            }
+            if (this.isExactTime) {
+              return DurationDropdownMap.EXACT_TIME;
+            }
+          }
+          return '';
+        },
+        set(duration) {
+          const update = {};
```

Comment:
Still not a huge fan of the complexity here in terms of the intensely nested if statements, and the use of `isSwitchingFromAllContentToCompleteDuration` and `currentDurationDropdown` state.

But let's get the tests in place so that we can refactor for simplification later down the line.